tools/resourceManagement/lib/resource/models/baseResource.js

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information. 
+
+'use strict';
+
+/**
+ * @class
+ * Initializes a new instance of the Resource class.
+ * @constructor
+ */
+class BaseResource {
+  constructor() { }
+
+  /**
+   * Defines the metadata of BaseResource
+   *
+   * @returns {object} metadata of BaseResource
+   *
+   */
+  mapper() {
+    return {
+      required: false,
+      serializedName: 'BaseResource',
+      type: {
+        name: 'Composite',
+        className: 'BaseResource',
+        modelProperties: {
+        }
```

Comment:
I don't know if this file is formatted, but shouldn't this be `: {}` instead of a line break?